core/app/models/spree/promotion_handler/promotion_duplicator.rb

```diff
@@ -9,8 +9,7 @@ def initialize(promotion, random_string: nil)
       def duplicate
         @new_promotion = @promotion.dup
         @new_promotion.path = "#{@promotion.path}_#{@random_string}"
```

Comment:
I've a feeling that a 'stronger' randomization might be required for the path - certainly a point for consideration.